core/modules/catalog-next/store/category/actions.ts

```diff
@@ -127,20 +128,27 @@ const actions: ActionTree<CategoryState, RootState> = {
     return CategoryService.getCategories(categorySearchOptions)
   },
   async loadCategories ({ commit, getters }, categorySearchOptions: DataResolver.CategorySearchOptions): Promise<Category[]> {
-    const searchingByIds = categorySearchOptions && categorySearchOptions.filters && categorySearchOptions.filters.id
+    const searchingByIds = !(!categorySearchOptions || !categorySearchOptions.filters || !categorySearchOptions.filters.id)
     const searchedIds: string[] = searchingByIds ? (categorySearchOptions.filters.id as string[]) : []
-    if (searchingByIds) { // removing from search query already loaded categories
+    const loadedCategories: Category[] = []
+    if (searchingByIds) { // removing from search query already loaded categories, they are added to returned results
+      for (const [categoryId, category] of Object.entries(getters.getCategoriesMap)) {
+        if (searchedIds.includes(categoryId)) {
+          loadedCategories.push(category as Category)
+        }
+      }
       categorySearchOptions.filters.id = searchedIds.filter(categoryId => !getters.getCategoriesMap[categoryId] && !getters.getNotFoundCategoryIds.includes(categoryId))
     }
     if (!searchingByIds || categorySearchOptions.filters.id.length) {
+      categorySearchOptions.filters = Object.assign({}, config.entities.category.filterFields ? config.entities.category.filterFields : {}, categorySearchOptions.filters)
```

Comment:
```suggestion
      categorySearchOptions.filters = Object.assign(config.entities.category.filterFields, categorySearchOptions.filters)
```
because `config.entities.category.filterFields` has `{}` by default